### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by zbahati**
Add unit tests for all the methods in all the classes:
-Book
-Person
-Student
-Teacher
-Classroom
-Rental
- all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: test/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    json_data = person.to_json
+    expect(json_data).to be_a(Hash)
+    expect(json_data[:age]).to eq(25)
+    expect(json_data[:name]).to eq('John')
+    expect(json_data[:parent_permission]).to be(true)
+    expect(json_data[:id]).to be_an(Integer)
+  end
+
+  it 'parses JSON data into a Person object' do
+    json_data = {
```
Comment (ghost) [inline test/person_spec.rb:45]: - [ ] Thank you for your hard work on this PR! 🙌 It's looking great, and I can see the effort you've put into it. 🌟Before we merge this PR, let's make sure all the methods in your classes get tested! 🧐 Please take a moment to thoroughly test each method to ensure everything is working smoothly. 🛠️. In this `Person class` the  `rent_book` method is not tested.

File: test/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  describe '#create_book' do
+    it 'adds a book to the library without rentals' do
+      book = library_manager.create_book('Patient & Good luck', 'Brymo')
+      expect(book.title).to eq('Patient & Good luck')
+      expect(book.author).to eq('Brymo')
+      expect(book.rentals).to be_empty
+    end
+  end
+end
```
Comment (ghost) [inline test/book_spec.rb]: - [ ] in the `Book class`, the `add rental ` method is not tested

File: test/classroom_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+describe Classroom do
+  let(:classroom) { Classroom.new('Class A') }
+
+  describe '#initialize' do
+    it 'creates a classroom with the given label' do
+      expect(classroom.label).to eq('Class A')
+      expect(classroom.students).to be_empty
+    end
+  end
+end
```
Comment (ghost) [inline test/classroom_spec.rb]: - [ ] In the `Classroom class`, the ` add_student` method is not tested

File: test/teacher_spec.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+  describe '#create_person' do
+    it 'adds a teacher to the library' do
+      teacher = library_manager.create_person(35, 'Mr. Smith', 'Math')
+      expect(teacher).to be_a(Teacher)
+      expect(teacher.age).to eq(35)
+      expect(teacher.name).to eq('Mr. Smith')
+    end
+  end
+end
```
Comment (ghost) [inline test/teacher_spec.rb]: - [ ]  `can_use_services?` method is not tested in your teachers class

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi  Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ Organized work
✅ Some methods are tested and are passing👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @zbahati and @Errytagedesign,

## Highlights!
- Good code organization and passing tests ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**